### PR TITLE
[NAN-676] enforce cap when activating a script

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Builds on https://github.com/NangoHQ/nango/pull/1950

## Issue ticket number and link
NAN-676
## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/constants.ts [hunk 1]
```diff
@@ -1,2 +1,4 @@
 export const SYNC_TASK_QUEUE = 'nango-syncs';
 export const WEBHOOK_TASK_QUEUE = 'nango-webhooks';
+//export const CONNECTIONS_WITH_SCRIPTS_CAP_LIMIT = 3;
+export const CONNECTIONS_WITH_SCRIPTS_CAP_LIMIT = Infinity;
```
Comment (TBonnin) [inline packages/shared/lib/constants.ts:4]: is that on purpose?

File: packages/webapp/src/pages/Integration/components/EnableDisableSync.tsx [hunk 2]
```diff
@@ -96,9 +113,26 @@ export default function EnableDisableSync({
...
+                setModalTitleColor('text-white');
+                setModalTitle('You’ve reached your connections limit!');
+                setModalContent(
+                    `Scripts are a paid feature. You can only use them with 3 connections or less.
+                    Upgrade or delete some connections to activate this script.`
+                );
+                setModalOkButtonTitle('Upgrade');
+                setModalCancelButtonTitle('Learn more');
+                setModalOkButtonLink('https://nango.dev/chat');
+                setModalCancelButtonLink('https://docs.nango.dev/REPLACE-ME');
```
Comment (TBonnin) [inline packages/webapp/src/pages/Integration/components/EnableDisableSync.tsx:127]: is it going to be replace later?

**Review by TBonnin — COMMENTED** [review]
question about the comment out limit constant but looks good otherwise

File: packages/shared/lib/constants.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/lib/constants.ts:4]: Yes, very much so. We don't want to cap until we can set the account information correctly via the API

File: packages/webapp/src/pages/Integration/components/EnableDisableSync.tsx (hunk 2 above)
Comment (khaliqgant) [reply packages/webapp/src/pages/Integration/components/EnableDisableSync.tsx:127]: Yup, launch blocker to replace.

File: packages/shared/lib/services/connection.service.ts [hunk 3]
```diff
@@ -990,6 +991,25 @@ class ConnectionService {
         }
     }
 
+    public async shouldCapUsage({ providerConfigKey, environmentId }: { providerConfigKey: string | undefined; environmentId: number }): Promise<boolean> {
```
Comment (bodinsamuel) [inline packages/shared/lib/services/connection.service.ts]: Can it really be undefined?

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/services/connection.service.ts (hunk 3 above)
Comment (khaliqgant) [reply packages/shared/lib/services/connection.service.ts]: I'll have to look again at it. It probably _shouldn't be_

Comment (khaliqgant) [reply packages/shared/lib/services/connection.service.ts]: Thanks, updated.

File: packages/shared/lib/constants.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/shared/lib/constants.ts:4]: Just thought about it during the night (could not sleep). We also have to think about enterprise/community 👌🏻

**Pull request merged**